### PR TITLE
MR paraphrase

### PR DESCRIPTION
**Pull request by msobrevillac**
This perturbation generates several references from a base reference by using a paraphrase generation model, keeping the key-value meaning representation (MR) not affected. Several realizations for a same MR are useful in NLG tasks, this way, this transformation can help this kind of tasks.

### CODE REVIEW

File: transformations/mr_paraphrase/transformation.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    
+        outputs = []
+    
+        input_phrase = "paraphrase: " + reference
+        input_ids = self.tokenizer.encode(input_phrase, return_tensors='pt')
+
+        preds = self.model.generate(
+            input_ids,
+            do_sample=True, 
+            max_length=len(reference.split()) + 10,
```
Comment (richplant) [inline transformations/mr_paraphrase/transformation.py:33]: Why the magic number here?

**Review by richplant — CHANGES_REQUESTED** [review]
I'm not convinced this transformation is entirely novel. There are several existing paraphrase transforms already logged (e.g. #94 #227 #253).

Also the build is failing since the test cases fail.

**Comment by marco-digio** [comment]
Hi @msobrevillac 
I was assigned as one of your reviewers. 
Thanks for this contribution!

I completely agree with @richplant about this PR. The idea is very interesting but it is not completely original, probably very similar to other PRs. It could be useful a clear description of the PR to better understand the novelty. 

Correctness: All checks have failed
Interface: I think you have chosen the correct interface(s).
Applicable Tasks & Keywords: Not sure if the task is correct, missing keywords.
Specificity: The change is quite general. 
Novelty: Similar transformations are implemented in NL-Augmenter. However, a clear description of the uniqueness of this PR could highlight the novelty better. 
Adding New Libraries: transformers is used but no requirements.txt is added. 
Description: The README is not clear.
Data and code source: Missing references of pretrained model. 
Test Cases: 5 test cases have been added.
Evaluating Robustness: No robustness evaluation has been performed.
Languages other than English: No

File: transformations/mr_paraphrase/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# MR Paraphrase 🦎  + ⌨️ → 🐍
+This perturbation generates several references from a base reference by using a paraphrase generation model, keeping the key-value meaning representation (MR) not affected. Several realizations for a same MR are useful in NLG tasks, this way, this transformation can help this kind of tasks.
+
+
+Author name: Marco Antonio Sobrevilla Cabezudo
+Author email: msobrevillac@usp.br
+Author Affiliation: University of São Paulo
+
+## What type of a transformation is this?
+This transformation could augment the number of references for an specific meaning representation by using a paraphrase model. The model was the generated by the parrot paraphraser.
```
[the remark this answers is not in the document]
Comment (msobrevillac) [reply transformations/mr_paraphrase/README.md:10]: Thanks for the comments. I'll add the citation.

**Comment by msobrevillac** [comment]
Many thanks for your comments, @marco-digio  and @richplant . I agree with you. This could be useful in Data-to-text generation tasks as adding diverse realizations to a unique MR is better in this kind of tasks. Datasets such as WebNLG usually contains several realizations for each MR. That's why I proposed this one, however, other PRs that generate paraphrase seems to be good enough and can be applied to E2E tasks. Do you think that I should close this PR?

**Comment by AbinayaM02** [comment]
> Many thanks for your comments, @marco-digio and @richplant . I agree with you. This could be useful in Data-to-text generation tasks as adding diverse realizations to a unique MR is better in this kind of tasks. Datasets such as WebNLG usually contains several realizations for each MR. That's why I proposed this one, however, other PRs that generate paraphrase seems to be good enough and can be applied to E2E tasks. Do you think that I should close this PR?

Hi @msobrevillac: Just for the purpose of bookkeeping, I'm closing the PR based on the discussions. Please feel free to re-open it or create a new PR later when you want to implement the logic. Thanks!

**Pull request closed without merging**